app/src/main/java/com/lightricks/feedexercise/ui/MainActivity.kt

```diff
@@ -1,15 +1,91 @@
 package com.lightricks.feedexercise.ui
 
-import androidx.appcompat.app.AppCompatActivity
+import android.annotation.SuppressLint
 import android.os.Bundle
+import android.util.Log
+import androidx.appcompat.app.AppCompatActivity
 import com.lightricks.feedexercise.R
+import com.lightricks.feedexercise.database.Entity
+import com.lightricks.feedexercise.database.FeedDatabase
+import com.lightricks.feedexercise.network.FeedApiService
+import com.lightricks.feedexercise.network.FeedResponse
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.schedulers.Schedulers
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
+import retrofit2.converter.moshi.MoshiConverterFactory
+
 
 /**
  * This is the main entry point into the app. This activity shows the [FeedFragment].
  */
+@Suppress("NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
 class MainActivity : AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.main_activity)
+        loadData()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+        compositeDisposable?.clear()
+    }
+
+    private var compositeDisposable: CompositeDisposable? = null
+
+    private fun loadData() {
+        val moshi = Moshi.Builder()
+            .add(KotlinJsonAdapterFactory())
+            .build()
+        val requestInterface = Retrofit.Builder()
+            .baseUrl("https://assets.swishvideoapp.com/Android/demo/")
+            .addConverterFactory(MoshiConverterFactory.create(moshi))
+            .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
+            .client(OkHttpClient.Builder().build())
+            .build()
+            .create(FeedApiService::class.java)
+
+        compositeDisposable = CompositeDisposable()
+        compositeDisposable?.add(requestInterface.getMetadataList()
+            .subscribeOn(Schedulers.io()) //[1]
+            .observeOn(AndroidSchedulers.mainThread()) //[2]
+            .subscribe({ feedResponse ->
+                handleResponse(feedResponse)
+            }, { error ->
+                handleNetworkError(error)
+            })
+        )
+
+
     }
+
+    @SuppressLint("CheckResult")
+    fun handleResponse(json: FeedResponse) {
+        val database = FeedDatabase.getDatabase(this)
+        database.feedDao().insertList(jsonToEntity(json))
+            .subscribeOn(Schedulers.io())
```

Comment:
What you can do is to chain `getMetadataList()` which is `Single` with `insertList()` which is `Completable` and have them both run on the background thread. Look up how to use `flatMapCompletable`.